lib/term.js

```diff
@@ -1,42 +1,55 @@
 const rdf = require('@rdfjs/data-model')
 
-function term (value, type, datatype) {
-  if (datatype) {
-    datatype = rdf.namedNode(datatype.value || datatype.toString())
+function blankNode (value) {
+  if (value && typeof value !== 'string') {
+    throw new Error('unknown type')
   }
 
-  if (type) {
-    if (type === 'BlankNode') {
-      return rdf.blankNode(value && value.toString())
-    }
+  return rdf.blankNode(value && value.toString())
+}
+
+function literal (value, languageOrDatatype) {
+  if (typeof value !== 'string' && typeof value !== 'number') {
```

Comment:
Boolean is also a valid literal.

```suggestion
  if (typeof value !== 'string' && typeof value !== 'number' && typeof value !== 'boolean') {
```